ludwig/schema/preprocessing.py

```diff
@@ -0,0 +1,699 @@
+class H3PreprocessingConfig(schema_utils.BaseMarshmallowConfig):
+
+    missing_value_strategy: Optional[str] = schema_utils.StringOptions(
+        MISSING_VALUE_STRATEGY_OPTIONS,
+        default="fill_with_const",
+        allow_none=False,
+        description="What strategy to follow when there's a missing value in an h3 column",
+    )
+
+    fill_value: Optional[Any] = schema_utils.PositiveInteger(
+        default=576495936675512319,
+        allow_none=False,
+        description="The value to replace missing values with in case the missing_value_strategy is fill_with_const",
+    )
+
+
+@dataclass
+@register_preprocessor(DATE)
+class DatePreprocessingConfig(schema_utils.BaseMarshmallowConfig):
+
+    missing_value_strategy: Optional[str] = schema_utils.StringOptions(
+        MISSING_VALUE_STRATEGY_OPTIONS,
+        default="fill_with_const",
+        allow_none=False,
+        description="What strategy to follow when there's a missing value in a date column",
+    )
+
+    fill_value: Optional[Any] = schema_utils.String(
+        default="",
+        allow_none=False,
+        description="The value to replace missing values with in case the missing_value_strategy is fill_with_const",
+    )
+
+    datetime_format: Optional[str] = schema_utils.String(
+        default=None,
+        allow_none=True,
+        description="This parameter can either be a datetime format string, or null, in which case the datetime "
+        "format will be inferred automatically.",
+    )
+
+
+@dataclass
+@register_preprocessor(VECTOR)
+class VectorPreprocessingConfig(schema_utils.BaseMarshmallowConfig):
+
+    vector_size: Optional[int] = schema_utils.PositiveInteger(
+        default=None,
+        allow_none=True,
+        description="The size of the vector. If None, the vector size will be inferred from the data.",
+    )
+
+    missing_value_strategy: Optional[str] = schema_utils.StringOptions(
+        MISSING_VALUE_STRATEGY_OPTIONS,
+        default="fill_with_const",
+        allow_none=False,
+        description="What strategy to follow when there's a missing value in a vector column",
+    )
+
+    fill_value: Optional[Any] = schema_utils.String(
+        default="",
+        allow_none=False,
+        pattern=r"^([0-9]+(\.[0-9]*)?\s*)*$",
+        description="The value to replace missing values with in case the missing_value_strategy is fill_with_const",
+    )
+
+
+def PreprocessingDataclassField(feature_type: str):
+    """Custom dataclass field that when used inside a dataclass will allow the user to specify a preprocessing
+    config.
+
+    Returns: Initialized dataclass field that converts an untyped dict with params to a preprocessing config.
+    """
+
+    class PreprocessingMarshmallowField(fields.Field):
+        """Custom marshmallow field that deserializes a dict for a valid preprocessing config from the
+        preprocessing_registry and creates a corresponding `oneOf` JSON schema for external usage."""
```

Comment:
Nit: update this docstring to remove `oneOf` ref.

Reply:
Done!